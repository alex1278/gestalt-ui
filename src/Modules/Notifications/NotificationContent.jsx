import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import styled, { withTheme } from 'styled-components';
import { FontIcon } from 'react-md';
import { Button } from 'components/Buttons';

const colors = {
  error: {
    color: '$md-red-700',
    icon: 'error',
  },
  warning: {
    color: '$md-amber-700',
    icon: 'warning',
  },
  info: {
    color: '$md-blue-700',
    icon: 'info',
  },
  success: {
    color: '$md-green-700',
    icon: 'check_circle',
  },
};

const Message = styled.div`
  display: flex;
  align-items: center;
  flex-wrap: wrap;
  margin: 8px;
  background-color: ${props => (props.status ? props.theme.colors[colors[props.status].color] : 'rgb(49, 49, 49)')};
  width: 344px;
  min-height: 48px;
  border-radius: 3px;
  padding: 6px 16px;
  pointer-events: initial;
  box-shadow: 0 3px 5px -1px rgba(0, 0, 0, 0.2), 0 6px 10px 0 rgba(0, 0, 0, 0.14), 0 1px 18px 0 rgba(0, 0, 0, 0.12);
`;

const Icon = styled(FontIcon)`
  padding-right: 8px;

  svg,
  i {
    color: white !important;
  }
`;

const Text = styled.div`
  color: white;
  display: flex;
  align-items: center;
  flex: 1 0 264px;

  svg,
  i {
    color: white !important;
  }
`;

const Action = styled.div`
  display: flex;
  align-items: center;
  color: white;
  justify-content: flex-end;

  button {
    svg,
    i {
      color: white !important;
    }
  }
`;

class NotificationContent extends PureComponent {
  static propTypes = {
    onRemove: PropTypes.func.isRequired,
    message: PropTypes.object.isRequired,
  };

  onRemove = () => {
    const { message, onRemove } = this.props;

    onRemove(message);
  }

  render() {
    const { message } = this.props;

    return (
      <Message status={message.status}>
        <Text>
          {message.icon && message.status && <Icon>{colors[message.status].icon}</Icon>}
          {message.message}
        </Text>

        <Action>
          <Button id={`close-${message.id}`} icon onClick={this.onRemove}>close</Button>
        </Action>
      </Message>
    );
  }
}

export default withTheme(NotificationContent);
