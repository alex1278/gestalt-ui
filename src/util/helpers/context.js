/**
 * generateContextEntityState
 * @param {*} params
 */
export function generateContextEntityState(params) {
  const entity = {
    id: params.environmentId || params.workspaceId || null,
    key: null,
  };

  if (params.workspaceId && params.environmentId) {
    entity.key = 'environments';
  }

  if (params.workspaceId && !params.environmentId) {
    entity.key = 'workspaces';
  }

  // if (params.workspaceId && params.environmentId && params.providerId) {
  //   entity.id = params.providerId;
  //   entity.key = 'providers';
  // }

  return entity;
}
