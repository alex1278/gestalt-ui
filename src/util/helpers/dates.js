import moment from 'moment';

export const getNow = () => moment();

export const getAgo = (num = 0, unit = 'day') =>
  moment().subtract(num, unit);

export const isRecentWithin = (timestamp, num, unit) =>
  moment(timestamp).isBetween(getAgo(num, unit), getNow());
