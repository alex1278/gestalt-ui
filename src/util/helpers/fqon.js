// sad hack for parsing url

export default function fqonParent(organization) {
  if (organization.org) {
    const urlParser = document.createElement('a');
    urlParser.href = organization.org.href;
    return urlParser.pathname.replace('/', '').split('/')[0];
  }

  return organization;
}
