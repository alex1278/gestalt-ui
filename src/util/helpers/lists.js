export function insertItem(array, payload) {
  return [...array, payload];
}

export function removeItem(array, payload) {
  const newArray = array.slice();
  newArray.splice(newArray.findIndex(a => a === payload), 1);

  return newArray;
}

export function removeItemById(array, id, keyField = 'id') {
  const newArray = array.slice();
  newArray.splice(newArray.findIndex(a => a[keyField] === id), 1);

  return newArray;
}

export function updateItem(array, payload) {
  return array.map((item) => {
    if (item.id === payload.id) {
      return {
        ...payload,
      };
    }

    // Leave every other item unchanged
    return item;
  });
}
