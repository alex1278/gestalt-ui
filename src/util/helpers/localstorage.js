/**
 * getItem
 * @param {*} key
 */
export const getItem = (key) => {
  try {
    return localStorage.getItem(key);
  } catch (err) {
    return undefined;
  }
};

/**
 * saveItem
 * @param {*} key
 * @param {*} value
 */
export const saveItem = (key, value) => {
  try {
    localStorage.setItem(key, value);
  } catch (err) {
    // ignore write errors
  }

  return undefined;
};

/**
 * removeItem
 * @param {*} key
 * @param {*} value
 */
export const removeItem = (key) => {
  try {
    localStorage.removeItem(key);
  } catch (err) {
    // ignore write errors
  }

  return undefined;
};
