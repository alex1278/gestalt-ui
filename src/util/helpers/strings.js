import { startCase, toLower } from 'lodash';

/**
 * getLastFromSplit
 * Parses and returns the last item in a structure like GrandPareng::Parent::Child
 * @param {String} string
 */
export function getLastFromSplit(string, delimiter = '::') {
  if (string) {
    const split = string.split(delimiter);

    return split[split.length - 1];
  }
  return string;
}

/**
 * truncate
 * @param {String} string
 * @param {Number} at
 * @param {String} trailing
 */
export function truncate(string, at, trailing = '...') {
  if (string) {
    if (string.length < at) {
      return string;
    }

    return `${string.substring(0, at).trim()}${trailing}`;
  }

  return string;
}

/**
 * toTitleCase
 * @param {String} string
 */
export function toTitleCase(string) {
  return startCase(toLower(string));
}


// TODO: Will be rmeoved when we have security config
export const checkIfPassword = field =>
  field && field.length && (field.toUpperCase().includes('PASSWORD') || field.toUpperCase().includes('SECRET') || field.toUpperCase().includes('KEY'));

export default {
  getLastFromSplit,
  truncate,
  toTitleCase,
  checkIfPassword
};

export const removeHostFromURL = (url) => {
  if (url) {
    return url.replace(/^.*\/\/[^/]+:?[0-9]?\//i, '');
  }

  return url;
};

export const isBase64 = (str) => {
  try {
    return btoa(atob(str)) === str;
  } catch (err) {
    return false;
  }
};
