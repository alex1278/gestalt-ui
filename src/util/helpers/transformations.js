import map from 'lodash/map';
import compact from 'lodash/compact';
import get from 'lodash/get';

/**
 * arrayToMap
 * @param {*} array
 * @param {*} keyName
 * @param {*} valueName
 */
export function arrayToMap(array = [], keyName = 'name', valueName = 'value') {
  if (!Array.isArray(array)) {
    return array;
  }

  return Object.assign({}, ...array.map(v => ({ [v[keyName]]: v[valueName] || '' })));
}

/**
 * mapTo2DArray
 * @param {*} object
 * @param {*} keyName
 * @param {*} valueName
 * @param {*} mergeSet
 */
export function mapTo2DArray(object = {}, keyName = 'name', valueName = 'value', mergeSet = {}) {
  if (typeof object !== 'object') {
    return object;
  }

  return map(object, (value, key) => (Object.assign({ [keyName]: key, [valueName]: value || '' }, mergeSet)));
}

/**
 * stringDemiltedToArray
 * convert comma delimited string to an array and remove blank entries
 * @param {*} string
 */
export function stringDemiltedToArray(string) {
  return compact(string.split(','));
}

/**
 * convertFromMaps
 * used to convert env vars from a map to an array and reconcile inherited vs own
 * @param {*} own
 * @param {*} inherited
 */
export function convertFromMaps(own = {}, inherited = {}, keyName = 'name', valueName = 'value') {
  const ownVars = mapTo2DArray(own, keyName, valueName, { inherited: false });

  // base case just return the mapped own vars
  if (!Object.keys(inherited)) {
    return ownVars;
  }

  const inheritedVars = mapTo2DArray(inherited, keyName, valueName, { inherited: true, overridden: false });

  const reconcileInheritedVars = inheritedVars.map((i) => {
    // find the dupe value in ownVars
    const index = ownVars.findIndex(o => o[keyName] === i[keyName]);

    // if the value is differnt mark the matching var as overriden
    if (index > -1 && ownVars[index][valueName] !== i[valueName]) {
      return { ...i, [valueName]: ownVars[index][valueName], inherited: true, overridden: true };
    }

    // default case the own env var
    return i;
  });

  const results = [...reconcileInheritedVars];

  // "merge" the ownVars back in
  ownVars.forEach((v) => {
    const index = inheritedVars.findIndex(o => o[keyName] === v[keyName]);

    if (index === -1) {
      results.push(v);
    }
  });

  return results;
}

/**
 * Builds an object from a delimited string
 * @param {String} string
 * @param {String} delimiter
 */
export function nestedObjectFromString(string, value = null, delimeter = '.') {
  const [last, ...paths] = string.split(delimeter).reverse();
  return paths.reduce((acc, el) => ({ [el]: acc }), { [last]: value });
}


export const getRoutePattern = (object, urlPattern) => {
  const keys = [];

  urlPattern.split('/').forEach((item) => {
    const search = item.startsWith(':');

    if (search) {
      keys.push(get(object, item.slice(1)));
    } else {
      keys.push(item);
    }
  });

  return keys.join('/');
};
