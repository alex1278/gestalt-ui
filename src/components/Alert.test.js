import React from 'react';
import { fireEvent } from 'react-testing-library';
import NotificationContent from './Alert';
import { renderWithTheme } from '../../test/helpers';

describe('(Notifications) NotificationContent', () => {
  it('renders to correct snapshot with default message payload', () => {
    const mockMessage = { id: 1, message: 'morty' };
    const { container } = renderWithTheme(
      <NotificationContent message={mockMessage} />
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders to correct snapshot when message has icon prop set to true', () => {
    const mockMessage = { id: 1, message: 'morty', icon: true };
    const { container } = renderWithTheme(
      <NotificationContent message={mockMessage} />
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders to correct snapshot when message status is info', () => {
    const mockMessage = { id: 1, message: 'morty', status: 'info' };
    const { container } = renderWithTheme(
      <NotificationContent message={mockMessage} />
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders to correct snapshot when message status is warning', () => {
    const mockMessage = { id: 1, message: 'morty', status: 'warning' };
    const { container } = renderWithTheme(
      <NotificationContent message={mockMessage} />
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders to correct snapshot when message status is error', () => {
    const mockMessage = { id: 1, message: 'morty', status: 'error' };
    const { container } = renderWithTheme(
      <NotificationContent message={mockMessage} />
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders to correct snapshot when message status is success', () => {
    const mockMessage = { id: 1, message: 'morty', status: 'success' };
    const { container } = renderWithTheme(
      <NotificationContent message={mockMessage} />
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders to correct snapshot when width is provider', () => {
    const mockMessage = { id: 1, message: 'morty', status: 'success' };
    const { container } = renderWithTheme(
      <NotificationContent width="auto" message={mockMessage} />
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders to correct snapshot when raised is provider', () => {
    const mockMessage = { id: 1, message: 'morty', status: 'success' };
    const { container } = renderWithTheme(
      <NotificationContent raised message={mockMessage} />
    );

    expect(container.firstChild).toMatchSnapshot();
  });


  it('handles onRemove when called', () => {
    const mockMessage = { id: 1, message: 'morty' };
    const onRemoveMock = jest.fn();
    const { container } = renderWithTheme(
      <NotificationContent message={mockMessage} onRemove={onRemoveMock} />
    );

    const button = container.querySelector('#close-1');
    fireEvent.click(button);

    expect(onRemoveMock).toBeCalledWith(mockMessage);
  });
});
