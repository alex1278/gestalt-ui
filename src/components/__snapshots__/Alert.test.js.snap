// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Notifications) NotificationContent renders to correct snapshot when message has icon prop set to true 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  margin: 8px;
  background-color: rgb(49,49,49);
  width: 344px;
  min-height: 48px;
  border-radius: 3px;
  padding: 6px 16px;
  pointer-events: initial;
  opacity: 0.98;
}

.c1 {
  color: white;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex: 1 0 264px;
  -ms-flex: 1 0 264px;
  flex: 1 0 264px;
}

.c1 svg,
.c1 i {
  color: white !important;
}

<div
  class="c0"
  width="344px"
>
  <div
    class="c1"
  >
    morty
  </div>
</div>
`;

exports[`(Notifications) NotificationContent renders to correct snapshot when message status is error 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  margin: 8px;
  background-color: #f44336;
  width: 344px;
  min-height: 48px;
  border-radius: 3px;
  padding: 6px 16px;
  pointer-events: initial;
  opacity: 0.98;
}

.c1 {
  color: white;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex: 1 0 264px;
  -ms-flex: 1 0 264px;
  flex: 1 0 264px;
}

.c1 svg,
.c1 i {
  color: white !important;
}

<div
  class="c0"
  width="344px"
>
  <div
    class="c1"
  >
    morty
  </div>
</div>
`;

exports[`(Notifications) NotificationContent renders to correct snapshot when message status is info 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  margin: 8px;
  background-color: #2196f3;
  width: 344px;
  min-height: 48px;
  border-radius: 3px;
  padding: 6px 16px;
  pointer-events: initial;
  opacity: 0.98;
}

.c1 {
  color: white;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex: 1 0 264px;
  -ms-flex: 1 0 264px;
  flex: 1 0 264px;
}

.c1 svg,
.c1 i {
  color: white !important;
}

<div
  class="c0"
  width="344px"
>
  <div
    class="c1"
  >
    morty
  </div>
</div>
`;

exports[`(Notifications) NotificationContent renders to correct snapshot when message status is success 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  margin: 8px;
  background-color: #00c853;
  width: 344px;
  min-height: 48px;
  border-radius: 3px;
  padding: 6px 16px;
  pointer-events: initial;
  opacity: 0.98;
}

.c1 {
  color: white;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex: 1 0 264px;
  -ms-flex: 1 0 264px;
  flex: 1 0 264px;
}

.c1 svg,
.c1 i {
  color: white !important;
}

<div
  class="c0"
  width="344px"
>
  <div
    class="c1"
  >
    morty
  </div>
</div>
`;

exports[`(Notifications) NotificationContent renders to correct snapshot when message status is warning 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  margin: 8px;
  background-color: #ff9800;
  width: 344px;
  min-height: 48px;
  border-radius: 3px;
  padding: 6px 16px;
  pointer-events: initial;
  opacity: 0.98;
}

.c1 {
  color: white;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex: 1 0 264px;
  -ms-flex: 1 0 264px;
  flex: 1 0 264px;
}

.c1 svg,
.c1 i {
  color: white !important;
}

<div
  class="c0"
  width="344px"
>
  <div
    class="c1"
  >
    morty
  </div>
</div>
`;

exports[`(Notifications) NotificationContent renders to correct snapshot when raised is provider 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  margin: 8px;
  background-color: #00c853;
  width: 344px;
  min-height: 48px;
  border-radius: 3px;
  padding: 6px 16px;
  pointer-events: initial;
  opacity: 0.98;
  box-shadow: 0 3px 5px -1px rgba(0,0,0,0.2),0 6px 10px 0 rgba(0,0,0,0.14),0 1px 18px 0 rgba(0,0,0,0.12);
}

.c1 {
  color: white;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex: 1 0 264px;
  -ms-flex: 1 0 264px;
  flex: 1 0 264px;
}

.c1 svg,
.c1 i {
  color: white !important;
}

<div
  class="c0"
  width="344px"
>
  <div
    class="c1"
  >
    morty
  </div>
</div>
`;

exports[`(Notifications) NotificationContent renders to correct snapshot when width is provider 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  margin: 8px;
  background-color: #00c853;
  width: auto;
  min-height: 48px;
  border-radius: 3px;
  padding: 6px 16px;
  pointer-events: initial;
  opacity: 0.98;
}

.c1 {
  color: white;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex: 1 0 264px;
  -ms-flex: 1 0 264px;
  flex: 1 0 264px;
}

.c1 svg,
.c1 i {
  color: white !important;
}

<div
  class="c0"
  width="auto"
>
  <div
    class="c1"
  >
    morty
  </div>
</div>
`;

exports[`(Notifications) NotificationContent renders to correct snapshot with default message payload 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  margin: 8px;
  background-color: rgb(49,49,49);
  width: 344px;
  min-height: 48px;
  border-radius: 3px;
  padding: 6px 16px;
  pointer-events: initial;
  opacity: 0.98;
}

.c1 {
  color: white;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex: 1 0 264px;
  -ms-flex: 1 0 264px;
  flex: 1 0 264px;
}

.c1 svg,
.c1 i {
  color: white !important;
}

<div
  class="c0"
  width="344px"
>
  <div
    class="c1"
  >
    morty
  </div>
</div>
`;
